nginx/nginx_test.go

```diff
@@ -80,10 +93,32 @@ func TestStopWaitsForGracefulShutdownOfNginx(t *testing.T) {
 	lb := newUpdaterWithBinary(tmpDir, "./fake_graceful_nginx.py")
 
 	assert.NoError(lb.Start())
+	assert.NoError(lb.Update(controller.IngressUpdate{Entries: []controller.IngressEntry{{
+		Host: "james.com",
+	}}}))
 	assert.NoError(lb.Stop())
 	assert.Error(lb.Health(), "should have waited for nginx to gracefully stop")
 }
 
+func TestNginxNotStartedUntilFirstUpdate(t *testing.T) {
+	tmpDir := setupWorkDir(t)
+	lb := newUpdaterWithBinary(tmpDir, "./fake_graceful_nginx.py")
+
+	assert.NoError(t, lb.Start())
+	assert.Nil(t, lb.nginx.Process)
+}
```

Comment:
For instance, we could use `fake_failing_nginx.sh` and make sure it only fails when the first update happens.